frappe/desk/form/save.py

```diff
@@ -17,9 +17,15 @@ def savedocs(doc, action):
 	doc.docstatus = {"Save": 0, "Submit": 1, "Update": 1, "Cancel": 2}[action]
 
 	if doc.docstatus == 1:
-		doc.submit()
+		if doc.bg_submit and doc.is_submittable:
```

Comment:
These properties are of `doc.meta` and not document itself. 

doc.meta contains the metaclass for a document

Reply:
I'm kinda thinking we should not give a user facing option for this - could lead to a lot of delays (unnecessary queuing of things even if they're not required which could block the queue for the jobs that *really* require queueing) (?) 🤔 

probably a (bool) class variable on the doctype controller (like `queue_on_submit` ?)